src/components/Frame/components/Loading/Loading.tsx

```diff
@@ -1,78 +1,69 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import debounce from 'lodash/debounce';
+import {useLazyRef} from '../../../../utilities/use-lazy-ref';
 import styles from './Loading.scss';
 
-export interface LoadingProps {}
-
-interface State {
-  progress: number;
-  step: number;
-  animation: number | null;
-}
-
 const INITIAL_STEP = 10;
 const STUCK_THRESHOLD = 99;
 
-export class Loading extends React.Component<LoadingProps, State> {
-  state: State = {
-    progress: 0,
-    step: INITIAL_STEP,
-    animation: null,
-  };
-
-  private ariaValuenow = debounce(() => {
-    const {progress} = this.state;
-    return Math.floor(progress / 10) * 10;
-  }, 15);
+export interface LoadingProps {}
 
-  componentDidMount() {
-    this.increment();
-  }
+export function Loading(_props: LoadingProps) {
+  const [progress, setProgress] = useState(0);
 
-  componentWillUnmount() {
-    const {animation} = this.state;
+  useEffect(() => {
+    let animation: number | undefined;
```

Comment:
Should animation, step currentProgress and previousProgress be refs? It seems to work ok in this closure but I'm not sure why.

Reply:
as we don't need those outside of `useEffect` I moved them all there, a race condition might occurs?

Reply:
> Should animation, step currentProgress and previousProgress be refs? It seems to work ok in this closure but I'm not sure why.

From the current implementation, it should work as-is from what I can tell. We're creating a closure, recursing and mutating the original values. `setProgress` will be consistently called with the correct values, and when the component is dismounted it'll have the current animation in scope to clear.